TACC-Cloud/geoapi
# task/WG-268: add authentification to backend

## Overview: ##

Add authentification to backend

## Related Jira tickets: ##

* [WG-268](https://tacc-main.atlassian.net/browse/WG-268)

## Summary of Changes: ##
* reworked env variables and there use in settings.py and in docker-compose ymls.
* added auth flow to be used by hazmapper and eventually taggit
* a pre-request check to refresh access token when it's expired or soon-to-be-expired

## Testing Steps: ##
1. To be tested with https://github.com/TACC-Cloud/hazmapper/pull/244
2. See https://github.com/TACC-Cloud/hazmapper/pull/244 for some testing ideas

## Notes: ##

TODO
- [x] save token/refresh-token in database 
- [x] consider that users could be using token not generated with the new auth flow (like they have a JWT but got it from somewhere else)
- [x] handle expiring refresh token. also handle if token needs to be refreshed during task (maybe assume all tasks take 2 hours? and just refresh if going to expire in 2hours?)
